Client/Frontend/UserContent/UserScripts/AllFrames/AtDocumentStart/RewardsReporting.js

```diff
@@ -0,0 +1,100 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
does being in AllFrames mean this runs in iframes? i think we probably only want it for top-level pages?

Reply:
Yes it does mean that, I'm not entirely sure actually if its required. Chromium's implementation has them overriding `OnBeforeRequest` (https://github.com/brave/brave-core/blob/5a217978eee7639e7d8b9b755a4af11504698dda/components/brave_rewards/browser/net/network_delegate_helper.cc#L80) and `ResourceDidLoad` (https://github.com/brave/brave-core/blob/5a217978eee7639e7d8b9b755a4af11504698dda/browser/brave_rewards/rewards_tab_helper.cc#L74)

So I assume it means that its executing for all frames

Reply:
maybe it needs to be in all iframes to deal with media embeds?

Reply:
actually can someone confirm that this does in fact need to run on all origins? would it be acceptable to only inject this content script on top-level twitch/youtube/vimeo/etc. pages?

Reply:
@diracdeltas as far as I know it checks the URL in the native libs before processing anything that is reported, however it'd be difficult to only inject this content script based on the page because of the available WebKit API's

Reply:
It would be nice to know though if this script can only be injected into main frame